ethash/src/compute.rs

```diff
@@ -141,7 +141,7 @@ pub fn quick_get_difficulty(header_hash: &H256, nonce: u64, mix_hash: &H256, pro
 			//
 			// This cannot be elided by the compiler as it doesn't know the implementation of
 			// `keccak_512`.
-			let mut buf: [u8; 64 + 32] = mem::uninitialized();
+			let mut buf: [u8; 64 + 32] = mem::MaybeUninit::uninit().assume_init();
```

Comment:
This is undefined behavior (as it was before), see https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initialization-invariant. The correct way to use `MaybeUninit` is to initialize the value to type `MaybeUninit<[u8; 64 + 32]>`, then actually initialize it (with `copy_nonoverlapping`) and then call `assume_init` to extract the underlying type (and ensure drop).